pandas/io/common.py

```diff
@@ -179,6 +179,7 @@ def get_filepath_or_buffer(filepath_or_buffer, encoding=None,
     filepath_or_buffer : a url, filepath (str, py.path.local or pathlib.Path),
                          or buffer
     encoding : the encoding to use to decode py3 bytes, default is 'utf-8'
+    mode : One of 'rb' or 'wb' or 'ab'. default: 'rb'
```

Comment:
@jreback thoughts on this parameter? I think it'd be better to just remove it, and hardcode mode='wb' in the call to `s3.get_filepath_or_buffer` down below. That's essentially what we do for URLs with the `BytesIO`.

Reply:
@jreback hard coding leads to failure to read from S3, an exception from s3fs https://github.com/dask/s3fs/blob/master/s3fs/core.py#L1005

I've decided to add it all the way in the call chain precisely for this reason. It might be possible to change s3fs implementation because from what I know S3 assets don't have a read/write notion in them, or split pandas code into 2 `get_readble_filepath_or_buffer` and `get_writable_filepath_or_buffer` but I don't feel I know the code base well enough judge.

Reply:
Ah, yes sorry I missed that.

In that case let's update the docstring to be numpydoc compliant: http://numpydoc.readthedocs.io/en/latest/format.html#sections

    mode : {'rb', 'wb', 'ab'}

Reply:
NP, my pleasure.